### PR TITLE
improve yt dlp player

### PR DESCRIPTION
**Pull request by float3**
continuation of #253

### CODE REVIEW

**Comment by pema99** [comment]
It's a little wack that the diff is basically useless here because we renamed the file simultaneously with changing a bunch of code. Do you plan to address that or nah?

**Comment by float3** [comment]
> It's a little wack that the diff is basically useless here because we renamed the file simultaneously with changing a bunch of code. Do you plan to address that or nah?

what do you suggest?

**Comment by pema99** [comment]
I think if you rename the file back to its original name, git might understand the diff. Worth a try, you can always change it in a later commit

**Comment by pema99** [comment]
So long as you have minimal changes in the file, git is made to understand when you are just renaming stuff, and it will keep the history

File: Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs [hunk 1]
```diff
@@ -1,203 +1,176 @@
...
-#endif
+                AudioSource audioSourceOutput = videoPlayer.GetTargetAudioSource(0);
+                if (audioSourceOutput != null)
+                    audioSourceOutput.volume = Mathf.Clamp01(volume);
             }
         }
     }
 
-    public static class YtdlpURLResolver
+    public class Request
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs:135]: lets change the name of this to something more descriptive

File: Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs [hunk 2]
```diff
@@ -1,203 +1,176 @@
...
+    public static class ytdlpURLResolver
     {
-        private static string _localYtdlpPath = Application.dataPath + "\\AudioLink\\yt-dlp.exe";
+        private static string _localytdlpPath = Application.dataPath + "\\AudioLink\\yt-dlp.exe";
 
         private static string _ytdlpPath = "";
         private static bool _ytdlpFound = false;
 
-        public static bool IsYtdlpAvailable()
+        public static bool IsytdlpAvailable()
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs:148]: Uppercase ytdlp here ? And in Locateytdlp?

File: Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs [hunk 3]
```diff
@@ -277,5 +259,160 @@ private static string LocateExecutable(string name)
...
+
+            bool hasVideoPlayer = _ytdlpPlayer.videoPlayer != null;
+            float playbackTime = hasVideoPlayer ? _ytdlpPlayer.GetPlaybackTime() : 0;
+            double videoLength = hasVideoPlayer ? _ytdlpPlayer.videoPlayer.length : 0;
+
+            using (new EditorGUI.DisabledScope(!available))
+            {
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    EditorGUILayout.LabelField(new GUIContent(" Video URL", EditorGUIUtility.IconContent("CloudConnect").image), GUILayout.Width(100));
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs:300]: " Video URL" space intentional?

Comment (pema99) [reply Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs:300]: looks like it

**Pull request merged**